lib/reused/option.js

```diff
@@ -0,0 +1,112 @@
+class Option {
+  static of(value) {
+    if (value instanceof Option)
+      return value;
+    if (value === null || value === undefined)
+      return none;
```

Comment:
I think this will fail the linter but I'll take a look at it.

Reply:
(use before declaration)

Reply:
I run eslint with the current rules definition and it didn't complain... Maybe I've got the rules wrong...

Reply:
then i am wrong!